netshare/generate/generate.py

```diff
@@ -14,14 +23,57 @@ def generate() -> None:
     2. Choose the best generated data and write it using the generate_api
     """
     model_generate()
-    denormalized_fields_dir = denormalize_fields()
-    chosen_data_dir = choose_best_model(denormalized_fields_dir)
-    copy_files(chosen_data_dir, get_generated_data_dir())
+    denormalize_fields()
+    choose_best_model()
 
 
-def choose_best_model(denormalized_fields_dir: str) -> str:
+def choose_best_model() -> None:
     """
     :return: the path to the data that was created by the chosen model
         (the raw data that should be shared with the user).
     """
-    return denormalized_fields_dir
+    # Current naive way: for each hyperparameter set, output one synthetic dataset by aggregating last ckpt's data.
+    # TODO
+    # 1. trimming strategy: trim each chunk or trim the entire csv after aggregation
+    # 2. pick best model across hyperparameters/ckpts by metrics
+
+    configs, config_group_list = create_chunks_configurations(generation_flag=True)
+    for config_group_idx, config_group in enumerate(config_group_list):
+        syn_csvs = []
+        for config_idx in config_group["config_ids"]:
+            config = configs[config_idx]
+            csv_root_folder = config["eval_root_folder"].replace(
+                get_raw_generated_data_dir(), get_generated_data_dir()
+            )
+            csv_folder = os.path.join(
+                csv_root_folder,
+                f"chunk_id-{config['chunk_id']}",
+            )
+            max_iteration_id = max(
+                [
+                    int(os.path.splitext(f)[0].split("-")[1])
+                    for f in os.listdir(csv_folder)
+                    if f.endswith(".csv")
+                ]
+            )
+            last_iteration_csv = pd.read_csv(
+                os.path.join(csv_folder, f"iteration_id-{max_iteration_id}.csv")
+            )
+            syn_csvs.append(last_iteration_csv)
+        # write aggregated csvs (from chunks) to `best_generated` dir
+        syn_csv_filename = ",".join(
+            [f"{k}-{v}" for k, v in config_group.items() if k not in ["config_ids"]]
+            + [
+                f"{k}-{v}"
+                for k, v in config.items()
+                if f"{k}_expand" in config and k not in ["dataset"]
+            ]
```

Comment:
If I understand correctly, you wrote this logic 3 times.
maybe worth writing it in the API, because the logic in `model_configuration` is based on the logic here?
I'm really not sure here...